File: src/main/java/it/carloni/luca/aurora/utils/ColumnName.java
package it.carloni.luca.aurora.utils;

public enum ColumnName {

    // SPECIFICATION
    FLUSSO("flusso"),
    VERSIONE("versione"),
    DT_FINE_VALIDITA("dt_fine_validita"),
    DT_INIZIO_VALIDITA("dt_inizio_validita"),
    DT_INSERIMENTO("dt_inserimento"),
    DT_RIFERIMENTO("dt_riferimento"),
    ERROR_DESCRIPTION("descrizione_errore"),
    ROW_COUNT("row_count"),
    ROW_ID("row_id"),
    TS_INIZIO_VALIDITA("ts_inizio_validita"),
    TS_INSERIMENTO("ts_inserimento"),
    TS_FINE_VALIDITA("ts_fine_validita"),

    // LOOKUP
    LOOKUP_TIPO("tipo_lookup"),
    LOOKUP_ID("lookup_id"),
    LOOKUP_VALORE_ORIGINALE("valore_originale"),
    LOOKUP_VALORE_SOSTITUZIONE("valore_sostituzione");

    private final String name;

    public String getName() { return name; }

    ColumnName(String name) { this.name = name; }
}

File: src/main/scala/it/luca/aurora/utils/ColumnName.scala
package it.luca.aurora.utils

object ColumnName extends Enumeration {

  protected case class Val(name: String) extends super.Val

  import scala.language.implicitConversions
  implicit def valueToColumnNameVal(x: Value): Val = x.asInstanceOf[Val]

  // Mapping Specification and raw layer data
  val Flusso: Val = Val("flusso")
  val Versione: Val = Val("versione")
  val DtFineValidita: Val = Val("dt_fine_validita")
  val DtInizioValidita: Val = Val("dt_inizio_validita")
  val DtInserimento: Val = Val("dt_inserimento")
  val DtRiferimento: Val = Val("dt_riferimento")
  val ErrorDescription: Val = Val("descrizione_errore")
  val RowCount: Val = Val("row_count")
  val RowId: Val = Val("row_id")
  val TsInizioValidita: Val = Val("ts_inizio_validita")
  val TsInserimento: Val = Val("ts_inserimento")
  val TsFineValidita: Val = Val("ts_fine_validita")

  // Lookup
  val LookupTipo: Val = Val("tipo_lookup")
  val LookupId: Val = Val("lookup_id")
  val LookupValoreOriginale: Val = Val("valore_originale")
  val LookupValoreSostituzione: Val = Val("valore_sostituzione")

}
